File: src/worldpainterStubs/typescript/BorderSettings.ts
export interface BorderSettings {
    getCentreX(): number;
    setCentreX(centreX: number): void;
    getCentreY(): number;
    setCentreY(centreY: number): void;
    getSize(): number;
    setSize(size: number): void;
    getSafeZone(): number;
    setSafeZone(safeZone: number): void;
    getWarningBlocks(): number;
    setWarningBlocks(warningBlocks: number): void;
    getWarningTime(): number;
    setWarningTime(warningTime: number): void;
    getSizeLerpTarget(): number;
    setSizeLerpTarget(sizeLerpTarget: number): void;
    getSizeLerpTime(): number;
    setSizeLerpTime(sizeLerpTime: number): void;
    getDamagePerBlock(): number;
    setDamagePerBlock(damagePerBlock: number): void;
    getChangeNo(): number;
    equals(o: any): boolean;
    hashCode(): number;
    clone(): BorderSettings;
}

File: src/worldpainterStubs/java/BorderSettings.java
package org.pepsoft.worldpainter;

import java.io.Serializable;

public interface BorderSettings extends Serializable, org.pepsoft.util.undo.Cloneable<World2.BorderSettings> {
    int getCentreX();

    void setCentreX(int centreX);

    int getCentreY();

    void setCentreY(int centreY);

    int getSize();

    void setSize(int size);

    int getSafeZone();

    void setSafeZone(int safeZone);

    int getWarningBlocks();

    void setWarningBlocks(int warningBlocks);

    int getWarningTime();

    void setWarningTime(int warningTime);

    int getSizeLerpTarget();

    void setSizeLerpTarget(int sizeLerpTarget);

    int getSizeLerpTime();

    void setSizeLerpTime(int sizeLerpTime);

    float getDamagePerBlock();

    void setDamagePerBlock(float damagePerBlock);

    long getChangeNo();

    @Override
    boolean equals(Object o);

    @Override
    int hashCode();

    @Override
    World2Interface.BorderSettings clone();
}
